app/src/main/java/mozilla/org/webmaker/activity/Element.java

```diff
@@ -1,10 +1,78 @@
 package mozilla.org.webmaker.activity;
 
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.os.Environment;
+import android.util.Log;
 import mozilla.org.webmaker.R;
 import mozilla.org.webmaker.WebmakerActivity;
+import mozilla.org.webmaker.util.Image;
+
+import java.io.File;
+import java.util.UUID;
 
 public class Element extends WebmakerActivity {
+
+    public String mUuid = UUID.randomUUID().toString();
+    public File mFile = new File(Environment.getExternalStorageDirectory() + File.separator + mUuid + ".jpg");
+    protected final int CAMERA_REQUEST_CODE = 1888;
+    protected final int MEDIA_REQUEST_CODE = 1889;
+
     public Element() {
         super("element", R.id.element_layout, R.layout.element_layout, R.menu.menu_element);
     }
+
+    /**
+     * Dispatches a new image capture intent.
+     */
+    public void dispatchCameraIntent() {
+        Intent intent = Image.getCameraIntent(mFile);
+        startActivityForResult(intent, CAMERA_REQUEST_CODE);
+    }
+
+    /**
+     * Dispatches a new media gallery intent.
+     */
+    public void dispatchMediaIntent() {
+        Intent intent = Image.getMediaStoreIntent(mFile);
+        startActivityForResult(intent, MEDIA_REQUEST_CODE);
+    }
+
+    /**
+     * Result handler for the image capture activity.
+     *
+     * @param requestCode
+     * @param resultCode
+     * @param data
+     */
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, Intent data) {
+        Bitmap bitmap = null;
+
+        // Return if user exited without capturing / selecting an image
+        if (resultCode == 0) {
+            Log.v("DATAURI:RESULT", resultCode + "");
```

Comment:
we should probably file a ticket to do a pass on logs, there are quite a few hanging around